src/components/chip/index.tsx

```diff
@@ -0,0 +1,335 @@
+   */
+  dismissIcon?: ImageSourcePropType;
+  /**
+   * Dismiss style
+   */
+  dismissIconStyle?: StyleProp<ImageStyle>;
+  /**
+   * Dismiss container style
+   */
+  dismissContainerStyle?: StyleProp<ImageStyle>;
+}
+
+type Props = ChipProps & ViewProps & TouchableOpacityProps;
+
+/**
+ * @description: Chip component
+ * @extends: TouchableOpacity
+ * @extendslink: docs/TouchableOpacity
+ * @example: https://github.com/wix/react-native-ui-lib/blob/master/demo/src/screens/componentScreens/ChipScreen.tsx
+ */
+const Chip: React.FC<ChipProps> = ({
+  avatar,
+  backgroundColor,
+  borderRadius,
+  counterBasicUi,
+  counterLabel,
+  counterProps,
+  containerStyle,
+  onDismiss,
+  dismissColor,
+  dismissIcon,
+  dismissIconStyle,
+  dismissContainerStyle,
+  iconColor,
+  iconSource,
+  iconStyle,
+  label,
+  labelStyle,
+  labelColor,
+  onPress,
+  resetSpacings,
+  size,
+  useSizeAsMinimum,
+  testID,
+  ...others
+}) => {
+
+  const renderIcon = useCallback(() => {
+    return (
+      <Image
+      // @ts-ignore
+        source={iconSource}
+        tintColor={iconColor}
+        style={[iconStyle, getMargins('iconSource')]}
+        testID={`${testID}.icon`}
+      />
+    );
+  }, [iconColor, iconSource, iconStyle]);
+
+
+  const renderCounter = useCallback(() => {
+    return (
+      <Badge
+        label={counterLabel}
+        size={BADGE_SIZES.medium}
+        backgroundColor={counterBasicUi ? 'transparent' : Colors.red30}
+        style={[getMargins('counter')]}
+        labelStyle={counterBasicUi && {color: Colors.grey20, ...Typography.text90R}}
+        {...counterProps}
+        testID={`${testID}.counter`}
+      />
+    );
+  }, [counterLabel, counterProps, counterBasicUi]);
+
+  const renderOnDismiss = useCallback(() => {
+    return (
+      <TouchableOpacity
+        style={[getMargins('dismiss'), dismissContainerStyle]}
+        onPress={onDismiss}
+        hitSlop={{top: 10, bottom: 10, left: 10, right: 10}}
+        testID={`${testID}.dismiss`}
+      >
+        <Image
+        // @ts-ignore
+          source={dismissIcon}
+          tintColor={dismissColor}
+          style={[dismissIconStyle]}
+          accessibilityLabel="dismiss"
+          testID={`${testID}.dismiss`}
+        />
+      </TouchableOpacity>
+    );
+  }, [dismissContainerStyle, onDismiss, dismissIcon, dismissIconStyle]);
+
+  const renderAvatar = useCallback(() => {
+    return (
+      <Avatar
+        size={20}
+        {...avatar}
+        // @ts-ignore
+        containerStyle={[getMargins('avatar'), avatar.containerStyle]}
+        testID={`${testID}.avatar`}
+      />
+    );
+  }, [avatar]);
+
+  const renderLabel = useCallback(() => {
+    return (
+      <Text
+        text90M
+        numberOfLines={1}
+        // @ts-ignore
+        tintColor={labelColor}
+        style={[styles.label, getMargins('label'), labelStyle]}
+        testID={`${testID}.label`}
+      >
+        {label}
+      </Text>
+    );
+  }, [label]);
```

Comment:
labelColor and labelStyle should also be included in the `useCallback` dependencies since they affect the rendered output.